prime-dotnet-webapi/HttpClients/Org Book API/OrgBookClient.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+
+using Prime.HttpClients.PharmanetCollegeApiDefinitions;
+using System.Web;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using Microsoft.Extensions.Logging;
+
+namespace Prime.HttpClients
+{
+    public class OrgBookClient : BaseClient, IOrgBookClient
+    {
+        private readonly HttpClient _client;
+        private readonly ILogger _logger;
+
+        public OrgBookClient(HttpClient client,
+        ILogger<OrgBookClient> logger)
+            : base(PropertySerialization.CamelCase)
+        {
+            // Auth header and cert are injected in Startup.cs
+            _client = client ?? throw new ArgumentNullException(nameof(client));
+            _logger = logger;
+        }
+
+        public async Task<string> GetOrgBookSearchRecordAsync(string orgName)
+        {
+            string registrationId = null;
+            HttpResponseMessage response = null;
+            try
+            {
+                response = await _client.GetAsync($"https://www.orgbook.gov.bc.ca/api/v2/search/credential/topic/facets?name=${HttpUtility.UrlEncode(orgName)}");
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Exception: ${ex.Message} for orgName: ${orgName}");
+            }
+
+            if (response != null)
+            {
+                string searchResult = await response.Content.ReadAsStringAsync();
+
+                var d = JObject.Parse(searchResult);
+
+                try
```

Comment:
The `try` should wrap `var d = JObject.Parse(searchResult);` as well because from playing with OrgBook URL to simulate an exception condition, the response returned might not be parse-able JSON

Reply:
Update as suggested.